### ISSUE TITLE
Table contextual classes (table-danger etc) do not adapt to dark mode

### ISSUE THREAD

**Issue by Aileron79** [twbs/bootstrap#39272]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

While alerts perfectly adapt to dark mode and change color accordingly, the contextual table classes (table-danger, table-success, ...) don't - looks kind of ugly, and very bright on a dark background. See screenshot and pen below.

![image](https://github.com/twbs/bootstrap/assets/16278437/52f4f25b-ac02-49da-9f7c-12c92ef0bbb9)

### Reduced test cases

https://codepen.io/Aileron79/pen/dywgNgY

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.3.1

**Comment by julien-deramond** [comment]
Thanks for opening this issue @Aileron79 
Indeed this is not yet functional as mentioned in [Tables > Variants](https://getbootstrap.com/docs/5.3/content/tables/#variants)

> Heads up! Because of the more complicated CSS used to generate our table variants, they most likely won’t see color mode adaptive styling until v6.

**Comment by Aileron79** [comment]
Thanks, I did not see that - probably I checked an older version's documentation. Looking forward to v6 then.

**Comment by pragyamishra56** [comment]
@Aileron79  @julien-deramond  If you allow me to assign this issue

**Comment by emmahsax** [comment]
I'm not sure if my issue is what the comment in the docs means, but when I upgraded to `v5.3.2`, my tables (using the default variant), no longer change color at all based on the computer's color mode changing. It's like the default variant (no `table-X` class at all), just is the light mode 100% of the time.

I did get as far into debugging to learn that the CSS works as I expect in `v5.3.0-alpha3`, but then it starts to break in `v5.3.0`, and continues to be broken up until `v5.3.2`. Maybe this helps?

But the strange thing is their docs (found [here](https://getbootstrap.com/docs/5.3/content/tables/)) do have the default variant changing based on the system's mode changing. Even just copy-pasting their basic table example doesn't work:

* Literally copy-pasted this block into my page:
```html
<table class="table">
  <thead>
    <tr>
      <th scope="col">#</th>
      <th scope="col">First</th>
      <th scope="col">Last</th>
      <th scope="col">Handle</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row">1</th>
      <td>Mark</td>
      <td>Otto</td>
      <td>@mdo</td>
    </tr>
    <tr>
      <th scope="row">2</th>
      <td>Jacob</td>
      <td>Thornton</td>
      <td>@fat</td>
    </tr>
    <tr>
      <th scope="row">3</th>
      <td colspan="2">Larry the Bird</td>
      <td>@twitter</td>
    </tr>
  </tbody>
</table>
```
* Dark mode: 
![image](https://github.com/twbs/bootstrap/assets/7562793/3ca47f32-2ac5-43bd-8647-afb76adc9e42)
* Light mode:
![image](https://github.com/twbs/bootstrap/assets/7562793/14352971-e951-417e-8b54-97d47f2c0b15)

**Comment by julien-deramond** [comment]
@emmahsax I'm not sure to understand what the issue is here. I've created https://codepen.io/julien-deramond/pen/NWomqGo with the basic example you've copied and it works both with light and dark mode (by using `data-bs-theme`).

Since it doesn't seem to be related to the variants, could you please create first a [Q&A Discussion](https://github.com/orgs/twbs/discussions/categories/q-a) with maybe a CodePen in it? It seems to be rather a usage issue than a bug.

**Comment by emmahsax** [comment]
I agree it is a usage issue, since it seems to work on your codepen and in their docs. But I will start a Q&A discussion.

**Comment by mdo** [comment]
They do in v6.
